### PR TITLE
Migration guide for `ThemeData.indicatorColor` deprecation

### PR DESCRIPTION
**Pull request by TahaTesser**
Related to https://github.com/flutter/flutter/pull/155072
PR deprecating `ThemeData.indicatorColor` https://github.com/flutter/flutter/pull/160024

## Presubmit checklist

- [x] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by sfshaza2 — APPROVED** [review]
lgtm! Please go ahead and remove this PR from draft mode and land.

**Comment by sfshaza2** [comment]
I'm landing this, as the (quiet) 3.7 release happened yesterday.

File: src/content/release/breaking-changes/index.md
Comment (sfshaza2) [inline src/content/release/breaking-changes/index.md:67]:
```suggestion
[Component theme normalization]: /release/breaking-changes/component-theme-normalization
[Deprecate `ThemeData.dialogBackgroundColor` in favor of `DialogThemeData.backgroundColor`]: /release/breaking-changes/deprecate-themedata-dialogbackgroundcolor
```

File: src/content/release/breaking-changes/index.md
Comment (sfshaza2) [inline src/content/release/breaking-changes/index.md:67]:
```suggestion
[Material 3 Tokens Update in Flutter]: /release/breaking-changes/material-design-3-token-update
[Component theme normalization]: /release/breaking-changes/component-theme-normalization
```

**Comment by Piinks** [comment]
Please do not merge migration guides before the actual change lands. Thank you!
@TahaTesser maybe we should mark these as a draft going forward to prevent this?

**Comment by TahaTesser** [comment]
> Please do not merge migration guides before the actual change lands. Thank you! @TahaTesser maybe we should mark these as a draft going forward to prevent this?

Apologies, this was marked as draft. Next time I'll definitely update the title with `[Do not merge]` to make to more clear.

**Pull request merged**
